github/compliance/milestone.py

```diff
@@ -25,11 +25,20 @@ def find_or_create_sprint_milestone(
         return ms
 
     title = _milestone_title(sprint=sprint)
+
+    sprint_release_decision = delivery.model.Sprint.find_sprint_date_by_name(
+        sprint=sprint,
+        name='release_decision',
+    )
+
+    if not sprint_release_decision:
+        raise RuntimeError(f'sprint missing release-decision, {sprint=}')
+
     ms = repo.create_milestone(
         title=title,
         state='open',
         description=f'used to track issues for upcoming sprint {title}',
-        due_on=sprint.release_decision.isoformat(),
+        due_on=sprint_release_decision.value.isoformat(),
```

Comment:
ohhh.... this gives me some nice ideas :-)

considering all our sprint-entries are currently "named dates", we might make `SprintDate` actually an instance of `datetime.datetime` (could be done by using inheritance - however, might be a tad too much work to be worth it)

It saddens me that `github3.py's` repository.create_milestone (this is what `repo` is, right?) does not accept datetime.datetime-instances as an alternative to passing-in a str (or does it?) :-/

Reply:
`repo` is of type `github3.repos.Repository`, yes.

Unfortunately no support for `datetime.datetime` ([source](https://github.com/sigmavirus24/github3.py/blob/43aa2d617a7d91743d7ab93c87bc9a52e300cb72/src/github3/repos/repo.py#L1112)):
```
        :param str due_on:
            (optional), ISO 8601 formatted due date
```

---

As `create_milestone` does not support `datetime.datetime` as type for its `due_on` parameter I do not see much benefit of inheriting `datetime.datetime` into `SprintDate`.

Reply:
ahh, yes, you are correct :-/